packages/oracle-node/src/fetchers/BaseFetcher.ts

```
@@ -95,4 +96,34 @@ export abstract class BaseFetcher implements Fetcher {
     }
     return prices;
   }
+
+  protected extractPricesSafely<T>(
+    items: T[],
+    extractPriceFn: (item: T, pricesObj: PricesObj) => number | undefined,
+    extractId: (item: T) => string
+  ): PricesObj {
+    const pricesObj: PricesObj = {};
+    for (const item of items) {
+      let id: string = "unknown";
+      try {
+        id = extractId(item);
+        if (!id) {
+          throw Error("Could not extract id from response");
+        }
```

Maybe not give the `id` initial value because it could lead to the situation when the `if (!id)` check will return false for the `unknown` value when it should throw an error.

---

Ohh great cache, fixed!